### ISSUE TITLE
How to re-train new dataset to existing classes?

### ISSUE THREAD

**Issue by gomappi** [qqwweee/keras-yolo3#640]
Hi, everyone.
I trained my custom classes with own dataset.
It works fine.
But, sometime occurs miss detections because of  unknown(new) image patterns.

So I want to re-train existing classes with new images.
I should re-train with ex-model(weight) and whole ex-dataset and new(adding) dataset?
Or is there another way to train more quickly?

**Comment by SrikarNamburu** [comment]
If you use the pre-trained weights then u can just train it on new dataset

**Comment by gomappi** [comment]
@SrikarNamburu 
Thank you for your reply.
I re-trained new dataset on pre-trained weights and same classes.txt, anchors.txt.
New dataset includes only one class of  existing classes.

New model returns good detection for included one class, but returns bad detection for other  classes.
I wrong any re-train settings?

**Comment by SrikarNamburu** [comment]
I think you have to increase your training set for new classes and retrain. Also, the classes.txt should be modified with the new classes.

**Comment by gomappi** [comment]
@SrikarNamburu 
Thank you for your advice, but I'm a little confused. 

Pre-trained model(weights) contains 3 classes (eg. A,B,C).
I re-train with new dataset includes only one class (A).

 pre-trained.h5
 class.txt->A,B,C
 dataset: A-class data only

Then, I want to detect 3 classes(A,B,C) with new model. 

I don't add any new classes.
I want to improve accuracy rate of existing class(A).

...any help with this?

**Comment by SrikarNamburu** [comment]
Oh I thought you were adding new class. So in that case may be increasing training data may help.
Good luck

**Comment by gomappi** [comment]
@SrikarNamburu 
Thank you for your reply.
There is nothing for it but to do so.
I will try it.
